src/lcm/argmax.py

```diff
@@ -1,31 +1,45 @@
 import jax
 import jax.numpy as jnp
+from jax import Array
+from jax.typing import ArrayLike
 
 # ======================================================================================
 # argmax
 # ======================================================================================
 
 
-def argmax(a, axis=None, initial=None, where=None):
-    """Compute the argmax of a n-dim array along axis.
+def argmax(
+    a: ArrayLike,
```

Comment:
Why do we use `a` here and `data` in `segment_argmax`?

Reply:
We follow the naming convention used in JAX and Tensorflow. I believe they use `data` for these segment operations to highlight that the operations are not directly performed on the input array.